### ISSUE TITLE
Device Sync

### ISSUE THREAD

**Issue by hugomrdias** [web3-storage/w3-protocol#106]
we are thinking about device linking because in the beginning we were not thinking about accounts so it would be just delegate `*` from one single DID and that would allow access to anything.

With accounts agent can get delegations from multiple accounts from other agents or local created accounts.

The problem is:

- Agent1 creates account1 and device links to Agent2
- Agent2 now receives a delegation for account2 from RandomAgent
- Agent2 has access to account1 and account2
- Agent1 may need to update delegations from Agent2 back. So user has all devices in sync.

So maybe instead of device linking we need device sync.

**Comment by hugomrdias** [comment]
@Gozala

**Comment by Gozala** [comment]
I think there is a value having a consistent access to all accounts across devices, but I would say it's not mandatory or even always desired. For example my email client on working machine is logged into bunch of work accounts, while my mobile phone is logged into just one. Likewise I have different slack sessions across devices.

This is to suggest that it's not always that one would want to have access to all account across all devices. There are also some security considerations to which may affect what accounts user may have access to on which device.


But again it is clear that there is going to be cases where one would want to have an access to the set of accounts across multiple agents. I think we could develop such a system as layer above, the baseline possibly encoding and storing shared capabilities in IPLD itself and storing it in the shared account, which would be a solution at the data layer.

Alternatively we could define separate higher level protocol similar to how we did with `upload/*` stuff.

**Comment by hugomrdias** [comment]
i dont think it needs to be higher level or more complex, a simple import account button that connects to another device (with qr code, ws whatever) and ask the user which account to send is enough.

this flow could even have the seed-phrase import in it too, this would make very similar to how crypto wallet work

**Comment by hugomrdias** [comment]
the initial device link could be just the first import account flow and the user select which account to send instead of just send everything. That maps to yours thoughts in the previous comments.
